Source/Parallelization/WarpXComm.cpp

```diff
@@ -181,7 +181,11 @@ WarpX::UpdateAuxilaryDataStagToNodal ()
             // ParallelCopy from coarse level
             for (int i = 0; i < 3; ++i) {
                 IntVect ng = Btmp[i]->nGrowVect();
-                Btmp[i]->ParallelCopy(*Bfield_aux[lev-1][i], 0, 0, 1, ng, ng, cperiod);
+                // Guard cells may not be up to date beyond ng_FieldGather
+                const amrex::IntVect& ng_src = guard_cells.ng_FieldGather;
+                // Copy Bfield_aux to Btmp, using up to ng_src (=ng_FieldGather) guard cells from
+                // Bfield_aux and filling up to ng (=nGrow) guard cells in Btmp
+                Btmp[i]->ParallelCopy(*Bfield_aux[lev-1][i], 0, 0, 1, ng_src, ng, cperiod);
```

Comment:
One quick question, to make sure that I understand correctly: What happens if `ng` is larger than `ng_src`, namely if I want to fill more guard cells in `Btmp` than the up-to-date guard cells I want to use from `Bfield_aux`?

Reply:
Well if you want a reliable answer it's probably more a question for amrex people :D 
But my understanding is that in this situation the guard cells that correspond to a regular cell in another box can always be filled, even if `ng` > `ng_src`, precisely because the value exists in another box (and so decreasing `ng_src` should probably increase the number of MPI communications). On the other hand, the guard cells at the edge of the domain will probably not be updated beyond `ng_src` (unless domain is periodic).

Reply:
@WeiqunZhang can you confirm pls? :)

Reply:
Yes, it's correct.  For valid cells plus ng ghost cells in Btmp, if there are cells with the same index within the valid+ng_src cells in the source MultiFab, they will be copied to Btmp.  If there are no cells having the same index, the data in those Btmp cells will not be touched.

> (and so decreasing ng_src should probably increase the number of MPI communications).

I think it should decrease the amount of data being communicated.

Reply:
> I think it should decrease the amount of data being communicated.

Ah, ok. I assumed a lower `ng_src` meant there'd be less data available locally so more communications, but I don't have a clear view on how the copy is actually performed.